langserver/config.go

```diff
@@ -137,41 +154,91 @@ func readConfigFromENV() (*Config, error) {
 
 // DidChangeConfiguration is required by the protocol.Server interface.
 func (s *server) DidChangeConfiguration(ctx context.Context, params *protocol.DidChangeConfigurationParams) error {
-	langserverAddressConfigPath := []string{"promql", "url"}
+	if params == nil {
+		return nil
+	}
+	// nolint: errcheck
+	s.client.LogMessage(
+		s.lifetime,
+		&protocol.LogMessageParams{
+			Type:    protocol.Info,
+			Message: fmt.Sprintf("Received notification change: %v\n", params),
+		})
+
+	setting := params.Settings
+
+	// the struct expected is the following
+	// promql:
+	//   url: http://
+	//   interval: 3w
+	m, ok := setting.(map[string]map[string]string)
+	if !ok {
+		// nolint: errcheck
+		s.client.LogMessage(ctx, &protocol.LogMessageParams{
+			Type:    protocol.Error,
+			Message: fmt.Sprint("unexpected format of the configuration"),
+		})
+		return nil
+	}
+	config, ok := m["promQL"]
```

Comment:
Changing `promql` to `promQL` this will break existing VS Code configurations of people using the extension.

Reply:
ah yes indeed. I missed this change :/.